src/main/scala/scalismo/statisticalmodel/DiscreteLowRankGaussianProcess.scala

```diff
@@ -218,9 +218,11 @@ case class DiscreteLowRankGaussianProcess[D: NDSpace, DDomain[DD] <: DiscreteDom
   override def marginalLikelihood(td: IndexedSeq[(PointId, Value, MultivariateNormalDistribution)]): Double = {
     require(td.nonEmpty, "provide observations to calculate the marginal likelihood")
     val matIds = td.flatMap(t => t._1.id * outputDim until t._1.id * outputDim + outputDim)
-    val U = DenseMatrix.tabulate[Double](matIds.length, rank)((x, y) => basisMatrix(matIds(x), y))
+    val r = variance.data.count(_ > 1e-7)
```

Comment:
Sounds good. I think a 1e-10 is enough as for higher point count this is already modelling a very small deformation. This could still prove to be an issue for small shapes (contained in some eps-radius) with only few points. But for these a scaling change seems anyway a reasonable thing to do.